src/background/events.ts

```diff
@@ -32,6 +32,8 @@ export const emitter = createNanoEvents<{
   chainsUpdated: () => void;
   chainChanged: (chain: Chain, origin: string) => void;
   'ui:chainSelected': (chain: Chain) => void;
+  signingError: (message: string) => void;
+  networkError: (message: string) => void;
```

Comment:
Why do we need to separate these? Let's create a single `'globalError'` event

Reply:
which will have a "type" or something

Reply:
or "name"

Reply:
```suggestion
  globalError: ({ name: 'NetworkError' | 'SigningError', message: string }) => void;
```

Reply:
I used the `type` ✅

Reply:
updated to `name` :)